pkg/services/contexthandler/authproxy/authproxy.go

```diff
@@ -84,14 +84,19 @@ type Options struct {
 
 // New instance of the AuthProxy.
 func New(cfg *setting.Cfg, options *Options) *AuthProxy {
-	header := options.Ctx.Req.Header.Get(cfg.AuthProxyHeaderName)
-	return &AuthProxy{
+	auth := &AuthProxy{
 		remoteCache: options.RemoteCache,
 		cfg:         cfg,
 		ctx:         options.Ctx,
 		orgID:       options.OrgID,
-		header:      header,
 	}
+	auth.init()
```

Comment:
nit-ish: What benefit do we get from introducing a separate `init()` method? Why not do this directly in `New()`?

Reply:
Agree, not needed. Removed.